src/components/accounts/TransactionItem.tsx

```diff
@@ -0,0 +1,262 @@
+import { StyleSheet, View } from "react-native";
+
+import TimeAgo from 'react-timeago'
+
+import { AppText } from "../shared/AppText";
+import { Colors, Layout, Typography } from "../../styles";
+import { Transaction, TransactionType } from "../../models/Transaction";
+
+import IncomingIcon from '../../assets/images/incoming.svg';
+import OutgoingIcon from '../../assets/images/outgoing.svg';
+import { Sats } from "./Sats";
+
+interface Props {
+  transaction: Transaction;
+  blockchainHeight: number;
+}
+
+const DateText = (props) => <AppText style={styles.dateTime}>{props.children}</AppText>;
+
+export default function TransactionItem({
+  transaction,
+  blockchainHeight
+}: Props) {
+
+  const showTime = !! transaction.timestamp;
+  const timestamp = new Date(transaction.timestamp as Date);
+  const { blockHeight, memo } = transaction;
+
+  const confirmations = getConfirmations(blockchainHeight, blockHeight);
+  const confirmationsText = getConfirmationsText(confirmations);
+  const confirmationsColorStyle = getConfirmationsColorStyle(confirmations);
+
+  const timeFormatter = (value: number, unit: string, suffix: string) => {
+    if (unit === 'second') {
+        return 'less than a minute ' + suffix;
+    } else if (unit === 'minute' || unit === 'hour') {
+        return `${value} ${unit}${
+          value !== 1 ? 's' : ''
+      } ${suffix}`;
+    } else {
+      return `${formatTime(timestamp)} - ${formatDate(timestamp)}`;
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.icon}>
+        { transaction.type === TransactionType.Send && <OutgoingIcon /> }
+        { transaction.type === TransactionType.Receive && <IncomingIcon width={19} height={19} /> }
+      </View>
+      <View style={styles.details}>
+        <View style={styles.leftColumn}>
+          { showTime && <TimeAgo
+            date={timestamp}
+            component={DateText}
+            live={true}
+            formatter={timeFormatter}
+          /> }
+          <Sats
+            sats={transaction.type === TransactionType.Send ?
+              -transaction.sent :
+              transaction.received
+            }
+            currencyStyle={styles.currency}
+            satsStyle={[styles.sats, ! showTime && styles.satsNoTime]}
+            satsLabelStyle={styles.satsLabel}
+            usdStyle={styles.usd}
+            usdLabelStyle={styles.usdLabel}
+          />
+        </View>
+        <View style={styles.rightColumn}>
+          <AppText style={[styles.confirmations, confirmationsColorStyle]}>{ blockchainHeight && confirmationsText }</AppText>
```

Comment:
TransactionConfirmation can be a separate component with separate ui and logic

Reply:
Done